### PR TITLE
issue20

### PR DESCRIPTION
**Pull request by Abhishek723**
issue 20 resolved

### CODE REVIEW

**Review by nemish-betterhalf — APPROVED** [review]

**Comment by nemish-betterhalf** [comment]
@Abhishek723 merge this branch to master branch

File: swiggyclone/service/urls.py [hunk 1]
```diff
@@ -1,16 +1,31 @@
...
+
 
 urlpatterns = [
-    path('', PlaceOrder.as_view(), name='placeOrder'),
-    path('register/restaurent/', RestaurentRegistertionView.as_view(), name='restaurentRegistertionView'),
-    path('register/branch/', BranchRegistertionView.as_view(), name='branchRegistertionView'),
-    path('register/fooditem/', FoodItemRegistertionView.as_view(), name='foodItemRegistertionView')
+    path('/', include(router.urls)),
+    path('/', include(restaurents_router.urls)),
+    path('/', include(branches_router.urls)),
```
Comment (nemish-betterhalf) [inline swiggyclone/service/urls.py:30]: Try this. urlpatterns = router.urls + restaurents_router.urls + branches_router.urls

File: swiggyclone/service/views.py [hunk 2]
```diff
@@ -89,3 +88,57 @@ def post(self,request):
...
+        branch = Branch.objects.select_related('foodItem').get(id=pk)
+        for orderDiscription in orderDiscriptions:
+            orderFoodItem = orderDiscription['foodItem']
+            orderQuantity = orderDiscription['quantity']
+            try:
+                instance = FoodItem.filter(branch_id=orderedBranch, id=orderFoodItem)
+            except FoodItem.DoesNotExist as e:
+                return Response({'error':'Given fooditem not found'},status = 404)
+            if instance[0].quantity < orderQuantity:
+                return Response({'error':'Given fooditem not found'},status = 404)
```
Comment (nemish-betterhalf) [inline swiggyclone/service/views.py]: If possible, raise these errors from serializers

File: swiggyclone/service/views.py [hunk 3]
```diff
@@ -89,3 +88,57 @@ def post(self,request):
...
+        total_price = 0
+        for orderDiscription in orderDiscriptions:
+            orderFoodItem = orderDiscription['foodItem']
+            orderQuantity = orderDiscription['quantity']
+            instance = FoodItem.filter(branch_id=orderedBranch, id=orderFoodItem)
+            foodItemPrice = instance.first().price
+            finalQuantity = instance.first().quantity - orderQuantity
+            serializer = FoodItemSerializer(instance[0], data = {"quantity":finalQuantity}, partial = True)
```
Comment (nemish-betterhalf) [inline swiggyclone/service/views.py]: why instance[0]? why not instance.first()?

File: swiggyclone/service/urls.py (hunk 1 above)
Comment (Abhishek723) [reply swiggyclone/service/urls.py:30]: Okk.

File: swiggyclone/service/views.py (hunk 2 above)
Comment (Abhishek723) [reply swiggyclone/service/views.py]: Improving it next commit.

File: swiggyclone/service/views.py (hunk 3 above)
Comment (Abhishek723) [reply swiggyclone/service/views.py]: Okk

File: swiggyclone/service/views.py [hunk 4]
```diff
@@ -89,3 +88,57 @@ def post(self,request):
...
+        
+        return Response({"message":"Order Placed", "total Price":total_price}, status = 200)
+
+
+
+class FoodItemViewSet(viewsets.ModelViewSet):
+    queryset = FoodItem.objects.all()
+    serializer_class = FoodItemSerializer
+    def get_queryset(self):
+        return FoodItem.objects.filter(restaurent_id = self.kwargs['restaurent_pk'], branch_id = self.kwargs[branch_pk])
```
Comment (nemish-betterhalf) [inline swiggyclone/service/views.py]: @dhaval-mehta

**Comment by dhaval-mehta** [comment]
> issue 20 resolved

You can write:

Fixes: #20

**Pull request merged**